app/controllers/orders_controller.rb

```diff
@@ -44,6 +44,11 @@ def show
     facility_ability = Ability.new(session_user, @order.facility, self)
     @order.being_purchased_by_admin = facility_ability.can?(:act_as, @order.facility)
     @order.validate_order! if @order.new?
+
+    return if @order.has_valid_payment?
+    @order.errors.add(:base, "Account is not valid for the orderer")
+    @order.errors.add(:base, "Please Change Payment Source or remove any invalid products from the Cart before continuing")
+    flash[:error] = "There are errors in your order:<br>#{@order.errors.full_messages.join('<br>')}".html_safe
```

Comment:
I think we should say something like:
```
"The payment source is not valid for the orderer.<br>  Please (clear your cart) [link to clear cart] or (select a different payment source) [link to change payment source] to continue."
```
... and put this in a translation file (i18n).

Then we can reuse in the places we need it like:
```suggestion
    @order.errors.add(:base, t("account_invalid_for_orderer"))
    flash[:error] = "There are errors in your order:<br>#{@order.errors.full_messages.join('<br>')}".html_safe
```